### PR TITLE
Automatically name selections on creation

### PR DESCRIPTION
**Pull request by VictoriaBeilsten-Edmands**

### CODE REVIEW

File: client/component/selections.tsx [hunk 1]
```diff
@@ -602,12 +602,18 @@ export function pointsToSelection(
...
+  alpha: number,
+  counter?: number,
+  updateCounters?: (t: SelectionType) => void
 ): BaseSelection {
   console.debug('Points', selectionType, points);
   const s = createSelection(selectionType, [false, false], points);
   s.colour = colour;
   s.alpha = alpha;
+  if (counter !== undefined && updateCounters) {
+    s.name = `${selectionType}${counter}`;
```
Comment (PeterC-DLS) [inline client/component/selections.tsx:614]: You don't check if name already exists

File: client/component/SelectionComponent.tsx [hunk 2]
```diff
@@ -41,6 +41,23 @@ export function SelectionComponent(props: SelectionComponentProps) {
...
+    circle: 0,
+    ellipse: 0,
+    sector: 0,
+    unknown: 0,
+  });
+
+  function updateCounters(t: SelectionType) {
+    const newCounters = counters;
+    newCounters[t]++;
+    setCounters(newCounters);
```
[the remark this answers is not in the document]
Comment (PeterC-DLS) [reply client/component/SelectionComponent.tsx:58]: Ok, so what happens when you do `setCounter(...counters)`?

File: client/component/selections.tsx [hunk 3]
```diff
@@ -608,6 +609,19 @@ export function pointsToSelection(
   const s = createSelection(selectionType, [false, false], points);
   s.colour = colour;
   s.alpha = alpha;
+  const selectionNames = selections.map((s) => s.name);
+  let counter = 0;
```
Comment (PeterC-DLS) [inline client/component/selections.tsx:613]: Use `do .... while`

**Review by PeterC-DLS — APPROVED** [review]

File: client/component/SelectionComponent.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (VictoriaBeilsten-Edmands) [reply client/component/SelectionComponent.tsx]: `updateCounters` function removed.

**Pull request merged**